libsolutil/CommonData.h

```diff
@@ -343,6 +343,49 @@ void joinMap(std::map<K, V>& _a, std::map<K, V>&& _b, F _conflictSolver)
 	}
 }
 
+template<typename T>
+class UniqueVector
+{
+public:
+	std::vector<T> const& contents() const { return m_contents; }
+	size_t size() const { return m_contents.size(); }
+	bool empty() const { return m_contents.empty(); }
+	auto begin() const { return m_contents.begin(); }
+	auto end() const { return m_contents.end(); }
+	void clear() { m_contents.clear(); }
+	bool contains(T const& _value) const
+	{
+		return std::find(m_contents.begin(), m_contents.end(), _value) != m_contents.end();
+	}
+
+	void pushBack(T _value)
+	{
+		if (!contains(_value))
+			m_contents.emplace_back(std::move(_value));
+	}
+
+	void pushBack(UniqueVector<T> const& _values)
+	{
+		for (auto&& value: _values)
+			pushBack(value);
+	}
+
+	void removeAll(std::vector<T> const& _values)
+	{
+		for (auto const& value: _values)
+			m_contents.erase(std::find(m_contents.begin(), m_contents.end(), value));
+	}
+
+private:
+	std::vector<T> m_contents;
+};
+
+template<typename T>
+void swap(UniqueVector<T>& _lhs, UniqueVector<T>& _rhs)
+{
+	std::swap(_lhs.contents(), _rhs.contents());
+}
```

Comment:
I was just looking at this to consider using it for https://github.com/ethereum/solidity/pull/15262 and noticed: this swap function won't work, will it? ``contents`` is a const reference, you can't swap these things...
what the ``std::swap`` below will do is to use the implicit move constructors of ``UniqueVector``...

Reply:
Yeah, you're right, `contents()` should ideally just return a reference...

Reply:
I would have honestly expected a compilation error here, this is really insidious.

Reply:
I mean, it's currently unused - and C++ only type-checks on instantiation :-). And I mean, it's not unused without reason: we don't need it, so we might as well remove the swap - ``std::swap`` will just work based on moving regardless